### DISCUSSION TITLE
Clipboard

### DISCUSSION THREAD

**Discussion by joshcangit** [makedeb/prebuilt-mpr#175] [Issues]
### Package name

clipboard

### List any other notes that might aid in the inclusion of this package (Is the upstream/MPR package popular? Is the upstream well-known or from a reputable source?).

# [![Clipboard by Jackson Huff](https://github.com/Slackadays/Clipboard/raw/main/documentation/readme-assets/CBBanner.png)](https://getclipboard.app/)

[![builds](https://img.shields.io/github/actions/workflow/status/Slackadays/Clipboard/build-clipboard.yml?branch=main&label=BUILDS&style=for-the-badge)](https://github.com/Slackadays/Clipboard/actions) [![checks](https://img.shields.io/github/actions/workflow/status/slackadays/Clipboard/lint-clipboard.yml?branch=main&label=CHECKS&style=for-the-badge)](https://github.com/Slackadays/Clipboard/actions) [![tests](https://img.shields.io/github/actions/workflow/status/Slackadays/Clipboard/test-clipboard.yml?branch=main&label=TESTS&style=for-the-badge)](https://github.com/Slackadays/Clipboard/actions) [![coverage](https://img.shields.io/codecov/c/github/slackadays/Clipboard/main?style=for-the-badge&label=COVERAGE&token=RO7KDOZ6Q2)](https://app.codecov.io/gh/Slackadays/Clipboard)

The Clipboard project is an easy-to-use terminal clipboard manager with many useful features.

Yuck, ugh, and puke! We can do better.

Clipboard is _<ins>your</ins>_ time and effort saver to use **anytime, anywhere in the terminal.** Cut, copy, paste, add, remove, and make note of **anything** sitting in your path, just how you would on a desktop! Have a helper with a perfect memory by your side. You'll be sitting on a sunny beach sipping a papaya smoothie with the love of your life in no time. Ok, so maybe that last part won't actually happen, but you get the idea:

![Clipboard makes your life easy.](https://github.com/Slackadays/Clipboard/raw/main/documentation/readme-assets/ClipboardMakesYourLifeEasy.png)

![Clipboard Demo Video](https://github.com/Slackadays/Clipboard/raw/main/documentation/readme-assets/ClipboardDemo.gif)

<details><summary><b>Click here to see our exquisite features.</b></summary>

- **Cut, copy, or paste files, directories, text, data, or any other kind of information.**
- **Add, remove, or make note of whatever you hold with Clipboard.**
- **Store everything in an _infinite_ number of different containers at your disposal.**
- **Choose if your containers are temporary or totally persistent.**
- **Connect right with your regular desktop [which Clipboard probably supports.](https://github.com/Slackadays/Clipboard/wiki/GUI-Clipboard-Compat)**
- **Do all of this in style with Clipboard's beautiful design.**
- **Works great on anything that supports C++20, which is Linux, Windows, macOS, FreeBSD, OpenBSD, NetBSD, OpenIndiana, DragonFlyBSD, Haiku, and probably more.**
- **Speak español, português, or Türkçe? You're in luck because Clipboard's in these languages too.**
- **Love freedom? We've got your back because Clipboard's 100% free and open source under the GPLv3.**

</details>

### [All Except Windows](https://github.com/Slackadays/Clipboard/blob/main/src/install.sh)
```bash
curl -sSL https://github.com/Slackadays/Clipboard/raw/main/src/install.sh | sh
```
### [Windows (run as Administrator)](https://github.com/Slackadays/Clipboard/blob/main/src/install.ps1)
```powershell
(Invoke-WebRequest -UseBasicParsing https://github.com/Slackadays/Clipboard/raw/main/src/install.ps1).Content | powershell
```

---

### Premade Builds

[![Packaging status](https://repology.org/badge/vertical-allrepos/clipboard.svg)](https://repology.org/project/clipboard/versions)

You can also download Clipboard [directly from GitHub Actions.](https://nightly.link/Slackadays/Clipboard/workflows/main/main)

---

### Install Manually (you'll need CMake and C++20 support)

Get the latest release instead by adding `--branch 0.5.0` right after `git clone...`.

Change the system installation prefix by adding `-DCMAKE_INSTALL_PREFIX=/custom/prefix` to `cmake ..`.
```bash
$ git clone https://github.com/Slackadays/Clipboard 
$ cd Clipboard/build
$ cmake -DCMAKE_BUILD_TYPE=MinSizeRel ..
$ cmake --build .
$ cmake --install .
```

### Uninstall

Remove all the files in `install_manifest.txt`. If you're not using Windows, you can also do `xargs rm < install_manifest.txt`.

## How To Use

You can substitute `cb` for `clipboard` in all commands.

Add a number to the end of the action to choose which clipboard you want to use (the default is 0), or `_` to use a persistent clipboard.

### Examples

```
$ clipboard copy NuclearLaunchCodes.pdf
$ clipboard cut69 MyDirectory
$ cb note "Keep this a secret"
$ cb cp420 bar.conf NotAVirus.pdf.mp3.exe baz.txt
$ cb show420
$ cb
$ cb add69 SomeFile
$ clipboard clr
```

---

**Copy** &emsp; `clipboard ([--]copy|[-]cp)[(num)|_(id)] (file) [files]` or `(something) | clipboard [([--]copy|[-]cp)][(num)|_(id)]`

---

**Cut** &emsp; `clipboard ([--]cut|[-]ct)[(num)|_(id)] (file) [files]`

---

**Paste** &emsp; `clipboard ([--]paste|[-]p)[(num)|_(id)]` or `clipboard [([--]paste|[-]p][(num)|_(id)] | (something)` or `clipboard [([--]paste|[-]p)][(num)|_(id)] > (some file)`

---

**Add Contents** &emsp; `clipboard ([--]add|[-]ad)[(num)|_(id)] (file|text) [files]` or `(something) | clipboard [([--]add|[-]ad)][(num)|_(id)]`

---

**Remove Contents** &emsp; `clipboard ([--]remove|[-]rm)[(num)|_(id)] (regex) [regexes]` or `(some regex) | clipboard [([--]remove|[-]rm)][(num)|_(id)]`

---

**Show Contents** &emsp; `clipboard ([--]show|[-]sh)[(num)|_(id)]`

---

**Clear Contents** &emsp; `clipboard ([--]clear|[-]clr)[(num)|_(id)]`

---

**Set Note** &emsp; `clipboard ([--]note|[-]nt)[(num)|_(id)] (text)` or `(something) | clipboard [([--]note|[-]nt)][(num)|_(id)]`

---

**Show Note** &emsp; `clipboard ([--]note|[-]nt)[(num)|_(id)]`

---

**Show Help Message** &emsp; `clipboard (-h|[--]help)`

---

**Check Clipboard Status** &emsp; `clipboard`

### Flags

**`--clipboard (clipboard)`, `-c (clipboard)`** &emsp; Add this to choose which clipboard you want to use.

---

**`--fast-copy`, `-fc`** &emsp; Add this to use links when copying, cutting, or pasting. If you modify the items that you used with this flag, then the items you paste will have the same changes.

### Environment Variables

**`CI`** &emsp; Set this to anything to make Clipboard overwrite existing items without a user prompt when pasting. This variable is intended for Continuous Integration scripts where a live human is not present to make decisions.

---

**`CLIPBOARD_LOCALE`** &emsp; Set this to the locale that only Clipboard will use for its commands and output, like `en_US.UTF-8` or `es_DO.UTF-8`.

---

**`CLIPBOARD_TMPDIR`** &emsp; Set this to the directory that only Clipboard will use to hold the items you cut or copy into a temporary directory.

---

**`CLIPBOARD_PERSISTDIR`** &emsp; Set this to the directory that only Clipboard will use to hold the items you cut or copy into a persistent directory.

---

**`CLIPBOARD_ALWAYS_PERSIST`** &emsp; Set this to anything to make Clipboard always use persistent clipboards.

---

**`CLIPBOARD_NOGUI`** &emsp; Set this to anything to disable integration with GUI clipboards.

---

**`CLIPBOARD_SILENT`** &emsp; Set this to anything to disable progress and confirmation messages from Clipboard.

---

**`CLIPBOARD_THEME`** &emsp; Set this to the color theme that Clipboard will use. Choose between `light`, `darkhighcontrast`, `lighthighcontrast`, `amber`, and `green` (the default is `dark`).

---

**`FORCE_COLOR`** &emsp; Set this to anything to make Clipboard always show color regardless of what you set `NO_COLOR` to.

---

**`NO_COLOR`** &emsp; Set this to anything to make Clipboard not show any colors.

## Need Help?

Go to [the Clipboard Wiki](https://github.com/Slackadays/Clipboard/wiki) for more information, ask your questions in [Clipboard Discussions](https://github.com/Slackadays/Clipboard/discussions), or join [our Discord group](https://discord.gg/J6asnc3pEG)!

[![Clipboard Wiki](https://img.shields.io/badge/Docs-Wiki-green?style=for-the-badge)](https://github.com/Slackadays/Clipboard/wiki) [![Questions? Just ask](https://img.shields.io/badge/Questions%3F-Just%20Ask-red?style=for-the-badge)](https://github.com/Slackadays/Clipboard/discussions) [![Discord Support](https://img.shields.io/badge/CHAT-DISCORD-blue?style=for-the-badge)](https://discord.gg/J6asnc3pEG)

## Thank You!

Say thank you to all the contributors who have helped make Clipboard great[.](https://www.youtube.com/watch?v=yjdHGmRKz08)

If you're feeling generous, feel free to give us a :star:!

<p align="right"><sup><sub><code>cb copy haters && cb > /dev/null</code></sub></sup></p>

**Comment by hwittenborn** [comment]
This package would have to be added into the MPR before it could be included here @joshcangit. If you'd be willing to maintain the package, I could help you get it added into the MPR though, and then it could be added into here no problem.

**Reply by joshcangit** [reply]
Actually, I have changed my mind.
Tried it out but I prefer `wl-clipboard` since it is now working fine on Ubuntu `22.04`.

I'm thinking of recommending something else.
